R/dm_wrap.R

```diff
@@ -0,0 +1,109 @@
+
+# dm_wrap takes a dm and a table name, which needs to be a leaf, and produces a
+# dm with one less table
+# dm_unwrap takes a dm, a table, and key pairs and produces a dm with one more table
+
+table_graph_position <- function(dm) {
+  graph <- create_graph_from_dm(dm, directed = TRUE)
+  vertices <- igraph::V(graph)
+  n_children <- map_dbl(vertices, ~ length(igraph::neighbors(graph, .x, mode = 'in')))
+  n_parents <- map_dbl(vertices, ~ length(igraph::neighbors(graph, .x, mode = 'out')))
+  node_types <- set_names(rep_along(vertices, "intermediate"), names(vertices))
+  node_types[n_parents == 0 & n_children == 1] <- "terminal parent"
+  node_types[n_children == 0 & n_parents == 1] <- "terminal child"
+  node_types[n_children == 0 & n_parents == 0] <- "isolated"
+  node_types
+}
+
+#' wrap a table from a dm
+#'
+#' @param dm a dm
+#' @param table a table, it needs to have either no parent or no child, but not
+#'   both of these.
+#' @param into the table to wrap `table` into, optional as it can be guessed
+#'   from the foreign keys unambiguously but useful to be explicit.
+#' @param silent if not silent (the default), the code to unwrap will be printed
+#'
+#' @noRd
+dm_wrap <- function(dm, table, into = NULL, silent = FALSE) {
+  table_name <- dm_tbl_name(dm, {{ table }})
+  positions <- table_graph_position(dm)
+  position <- positions[table_name]
+  # so we can forward it
+  if (missing(into)) into <- missing_arg()
```

Comment:
The idiomatic way might be to quote-unquote and use `quo_is_null()` to check nullness.

Reply:
Thanks and done 👍